MRA-codegen.py

```diff
@@ -70,16 +70,25 @@ def __init__(self, component: str, verbosity: int = DEFAULT_VERBOSE_LEVEL):
         self.interface_folder = os.path.join(self.component_folder, 'interface')
         self.analyze()
 
+    def check_component_name_valid(self):
+        """Ensure component name is valid, for instance no dash character ... (causes problem with protoc --python_out)"""
+        if '-' in self.component:
+            raise Exception(f'illegal component name, no dash (-) allowed in {self.component}')
+
+    def check_component_interface_valid(self):
+        """Ensure component interface (.proto files) are valid."""
+        # TODO: warn in case other than all_interface_parts .proto files are detected?
```

Comment:
Perhaps check if other files are in list ["Local", "State", "Parameters"]. If not matching, then unexpected proto file.

Reply:
Well yeah that is what the TODO suggests... 

Gonna leave as is.

Reply:
agree with proposal to leave as it is.